pygments/lexers/berry.py

```diff
@@ -0,0 +1,99 @@
+import re
+from pygments.lexer import RegexLexer, words, default, include, bygroups
+from pygments.token import Text, Comment, Whitespace, Operator, Keyword, Name, String, Number, Punctuation
+
+__all__ = ['BerryLexer']
+
+line_re = re.compile('.*?\n')
+
+class BerryLexer(RegexLexer):
+
+    # For Berry <https://github.com/berry-lang/berry> source code (version v0.1.0)
+    # Contributed by Shaun Brown (Beormund) <https://github.com/Beormund>
+    
+    name = 'Berry'
+    aliases = ['berry', 'be']
+    filenames = ['*.be']
+    mimetypes = ['text/x-berry', 'application/x-berry']
+
+    _name = r'\b[^\W\d]\w*'
+
+    # (def)\b((?:\s)*)
+    # (def)\b(\s*$)(?<!\()
+    # (def)\b((?:\s)*)(?=[^\W\d]\w*)
+
+    tokens = {
+        'root': [
+            include('whitespace'),
+            include('numbers'),
+            include('keywords'),
+            (r'(def)\b((?:\s)*)' + '(?=' + _name + ')', bygroups(Keyword.Declaration, Text), 'funcname'),
```

Comment:
I'm struggling to get the revised regex to work for berry's anonymous function syntax: def (). So the regex needs to work for both def myfuncname(myparam) and def (myparam) which has no function name.

Reply:
`(rf'(def)(\s+)({_name})?, bygroups(Keyword.Declaration, Whitespace, Name.Function),`

Nam.Function may be empty. Is that OK?

Reply:
You can also split it in two token definitions, the one with the name first.

Reply:
>  `(rf'(def)(\s*)({_name})?', bygroups(Keyword.Declaration, Whitespace, Name.Function)),`



That has the problem that it will match `definition` or similar, since `\s*` matches the empty string and so does `({_name})?`. You need one of

- `rf'(def)\b(\s*)({_name})?`
- `rf'(def)(?:(\s+)({_name})|\b)`
- Two separate states as Georg suggested, with `rf'(def)(\s+)({_name})'` and `r'def\b'`. That is what I would find easiest.

Reply:
Two separate states created for def keyword:
rf'(def)(\s+)({_name})' (line 37)
r'def\b' (line 63)

Reply:
> (By the way, when you fix this problem, make sure it's covered in the test.)

I included a def definitive () in the test.